### ISSUE TITLE
[INTERNAL] Move <<, >>, ==,!=,<,>,>=,>= to operators

### ISSUE THREAD

**Issue by jfalcou** [jfalcou/eve#377]
Like for +-*/ etc, those basic versions of basic operators must move into details and let the function version handle all the complexity.

Doing so will make wide not depend on stupid stuff like is_nan etc...

**Comment by jfalcou** [comment]
Depends on #404

**Comment by jfalcou** [comment]
After the merge of #404, I think the comparison operators may not be required to be moved. Shifts should still. A rapid check of wide.hpp dependencies using find_dep still show some nasty dependencies on unwanted files but it looks like they're related to other files.

So I think we'll move << and >> to wide and consider this issue done.

**Comment by jfalcou** [comment]
This is now a duplicate of #435
